powersoftau/src/batched_accumulator.rs

```diff
@@ -461,28 +475,31 @@ impl<'a, E: Engine + Sync> BatchedAccumulator<'a, E> {
                     check_input_for_correctness,
                     &input,
                 )?;
-                assert_eq!(
-                    accumulator.tau_powers_g2.len(),
-                    0,
-                    "during rest of tau g1 generation tau g2 must be empty"
-                );
-                assert_eq!(
-                    accumulator.alpha_tau_powers_g1.len(),
-                    0,
-                    "during rest of tau g1 generation alpha*tau in g1 must be empty"
-                );
-                assert_eq!(
-                    accumulator.beta_tau_powers_g1.len(),
-                    0,
-                    "during rest of tau g1 generation beta*tau in g1 must be empty"
-                );
+
+                if !accumulator.tau_powers_g2.is_empty() {
+                    return Err(Error::InvalidLength {
+                        expected: 0,
+                        got: accumulator.tau_powers_g2.len(),
+                    });
+                }
+
+                if !accumulator.alpha_tau_powers_g1.is_empty() {
+                    return Err(Error::InvalidLength {
+                        expected: 0,
+                        got: accumulator.alpha_tau_powers_g1.len(),
+                    });
+                }
+
+                if !accumulator.beta_tau_powers_g1.is_empty() {
+                    return Err(Error::InvalidLength {
+                        expected: 0,
+                        got: accumulator.beta_tau_powers_g1.len(),
+                    });
+                }
 
                 tau_powers_g1.extend_from_slice(&accumulator.tau_powers_g1);
-                tau_powers_g2.extend_from_slice(&accumulator.tau_powers_g2);
```

Comment:
It should be safe to remove these since they were empty vectors

Reply:
I agree